### ISSUE TITLE
SC2154 conditionally complain about CAPITAL variables not defined

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#2019]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2154
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2154)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

_... kinda. The list is too big (~80 issues). And there are multiple kinda-duplicates._
#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
if [ "${SKIP_TERM_MODE}" -eq 0 ] ; then
	byobu-disable
else
	byobu-enable
fi
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

SC2154 for `SKIP_TERM_MODE`.

I know that capital variables have this shady status of sometimes being considered as "provided by the context". However, I'd propose a shellcheck flag, or a mode (`set -Eeuo pipefail`) that would propose assigning defaults `${SKIP_TERM_MODE:-}`.

Bonus points: In this case, a numerical value.

**Comment by koalaman** [comment]
Have a look at the `check-unassigned-uppercase` optional check from `shellcheck --list-optional`:

```sh
name:    check-unassigned-uppercase
desc:    Warn when uppercase variables are unassigned
example: echo $VAR
fix:     VAR=hello; echo $VAR
```

It does exactly this, and you can enable it with `shellcheck -o check-unassigned-uppercase ..` or `# shellcheck enable=check-unassigned-uppercase`
